src/main/java/saps/common/core/storage/swift/SwiftAPIClient.java

```diff
@@ -168,18 +169,20 @@ public void deleteFile(String containerName, String filePath) throws Exception {
   }
 
   public boolean existsTask(String containerName, String basePath, String taskId)
-      throws IOException {
-    List<String> files = this.listFiles(containerName, basePath);
-    for (String filePath : files) {
-      try {
-        if (Paths.get(filePath).getFileName().toString().equals(taskId)) {
-          return true;
-        }
-      } catch (NullPointerException np) {
-        LOGGER.error("Error while checking if task exists", np);
-        return false;
-      }
-    }
-    return false;
+      throws IOException {	  
+	  String url = String.format(TASK_URL_PATTERN, swiftUrl, containerName, basePath, taskId);
```

Comment:
do we still need the `listFiles` implementation? could we removed it?

Reply:
It is still used in other methods.